x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ssl/SSLService.java

```diff
@@ -446,6 +449,32 @@ private SSLConfiguration loadConfiguration(String key, Settings settings, Map<SS
         }
     }
 
+    private void validateServerConfiguration(String prefix) {
+        assert prefix.endsWith(".ssl");
+        SSLConfiguration configuration = getSSLConfiguration(prefix);
+        final String enabledSetting = prefix + ".enabled";
+        if (settings.getAsBoolean(enabledSetting, false) == true) {
+            // Client Authentication _should_ be required, but if someone turns it off, then this check is no longer relevant
+            final SSLConfigurationSettings configurationSettings = SSLConfigurationSettings.withPrefix(prefix + ".");
+            if (isConfigurationValidForServerUsage(configuration) == false) {
+                throw new ElasticsearchSecurityException("invalid configuration for " + prefix +
+                    " - a server context requires a key and certificate, but these have not been configured; either [" +
+                    configurationSettings.x509KeyPair.keystorePath.getKey() + "] or [" +
+                    configurationSettings.x509KeyPair.certificatePath.getKey() + "] should be set");
+            }
+        } else if (settings.hasValue(enabledSetting) == false) {
+            final List<String> sslKeys = settings.keySet().stream()
```

Comment:
```suggestion
            final List<String> sslConfigKeys = settings.keySet().stream()
```

to make it immediately obvious when looking at the code these are not SSL Keys but configuration keys